name.abuchen.portfolio/src/name/abuchen/portfolio/model/Transaction.java

```diff
@@ -419,6 +419,16 @@ public void removeUnit(Unit unit)
         this.updatedAt = Instant.now();
     }
 
+    /**
+     * Remove all units by unit type
+     */
+    public void removeUnits(Unit.Type type)
+    {
+        units = units == null ? new ArrayList<>() : units;
+        getUnit(type).ifPresent(units::remove);
```

Comment:
I think #getUnit returns the first unit of that type. Hence the method does not delete all units by this type, but only one (maybe the first - the order is not defined)

Reply:
Ops... mistake fixed :smile:
Thanks 👍🏻